app/Http/Controllers/RankingController.php

```diff
@@ -223,10 +223,15 @@ public function index($mode, $type)
 
     public function kudosu()
     {
+        static $maxResults = 1000;
+
+        $maxPage = $maxResults / static::PAGE_SIZE;
+        $page = min(get_int(request('page')) ?? 1, $maxPage);
+
         $scores = User::default()
             ->with('country')
             ->orderBy('osu_kudostotal', 'desc')
-            ->paginate(static::PAGE_SIZE, ['*'], 'page', null, 1000);
+            ->paginate(static::PAGE_SIZE, ['*'], 'page', $page, $maxResults);
```

Comment:
actually, doesn't `paginate` only have 4 params? 🤔

Reply:
that's what you think 👀

Reply:
oh it's a magic param :|